pkg/virt-controller/watch/vmi.go

```diff
@@ -2447,3 +2449,52 @@ func (c *VMIController) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 		vmi.Labels[virtv1.MemoryHotplugOverheadRatioLabel] = *overheadRatio
 	}
 }
+
+func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
+	if len(dvs) == 0 {
+		return
+	}
+
+	dvsReadyCondition := virtv1.VirtualMachineInstanceCondition{
+		Status:  k8sv1.ConditionTrue,
+		Type:    virtv1.VirtualMachineInstanceDataVolumesReady,
+		Reason:  virtv1.VirtualMachineInstanceReasonAllDVsReady,
+		Message: "All of the VMI's DVs are bound and not running",
+	}
+
+	dvmessages := []string{}
+	for _, dv := range dvs {
+		cstatus, cmessage := getReadyConditionStatus(dv.Status.Conditions)
+		if cstatus != k8sv1.ConditionTrue {
+			dvsReadyCondition.Reason = virtv1.VirtualMachineInstanceReasonNotAllDVsReady
+			if cmessage != "" {
+				dvmessages = append(dvmessages, cmessage)
+			}
+			if cstatus == k8sv1.ConditionFalse {
+				dvsReadyCondition.Status = cstatus
+			} else if cstatus == k8sv1.ConditionUnknown && dvsReadyCondition.Status == k8sv1.ConditionTrue {
+				dvsReadyCondition.Status = cstatus
+			}
+		}
+	}
+
+	if dvsReadyCondition.Status != k8sv1.ConditionTrue {
+		if len(dvmessages) > 0 {
+			dvsReadyCondition.Message = "Not all of the VMI's DVs are ready: " + strings.Join(dvmessages, ",")
```

Comment:
nit: to make things easier I would simplify this too in just always reporting a static `"Not all of the VMI's DVs are ready"`

Reply:
CDI is actually never reporting custom messages for the Ready condition (it's using custom messages only for `DataVolumeRunning` and `DataVolumeBound`). Still I think that this can make the codo future proof in the case CDI will start providing human readable messages.